transfer-sh.el

```diff
@@ -54,39 +54,69 @@
   :type '(string)
   :group 'transfer-sh)
 
+(defcustom transfer-sh-upload-agent-command
+  (cond
+   ((executable-find "curl")
+    "curl")
+   ((executable-find "wget")
+    "wget"))
+  "Command used to upload files to transfer.sh"
+  :type '(string)
+  :group 'transfer-sh)
+
+(defcustom transfer-sh-upload-agent-arguments
+  (cond
+   ((executable-find "curl")
+    (list "--silent" "--upload-file"))
+   ((executable-find "wget")
+    (list "--method" "PUT" "--output-document" "-"  "--no-verbose" "--quiet" "--body-file")))
+  "Suffix arguments to `transfer-sh-upload-agent-command'"
+  :group 'transfer-sh)
+
 ;;;###autoload
-(defun transfer-sh-upload-file-async (local-filename &optional remote-filename)
+(defun transfer-sh-upload-file-async (local-filename)
   "Uploads file LOCAL-FILENAME to transfer.sh in background.
 
-If no REMOTE-FILENAME is given, the LOCAL-FILENAME is used."
+Read from minibuffer the remote file name visible in the transfer.sh link and run `transfer-sh-run-upload-agent'."
   (interactive "ffile: ")
-  (async-start
-   `(lambda ()
-      ,(async-inject-variables "local-filename")
-      ,(async-inject-variables "remote-filename")
-      (shell-command-to-string
-        (concat "curl --silent --upload-file "
-                (shell-quote-argument local-filename)
-                " " (shell-quote-argument (concat "https://transfer.sh/" remote-filename)))))
-   `(lambda (transfer-link)
-      (kill-new transfer-link)
-      (message transfer-link))))
+  (let* ((remote-filename (read-from-minibuffer
```

Comment:
Here you could use `(or remote-filename (read-from-minibuffer...`.

Reply:
solved